src/lib.rs

```diff
@@ -280,6 +278,34 @@ pub trait Domain {
     /// Returns `None` if `out_plaintext` does not contain a valid byte encoding of an
     /// `EphemeralSecretKey`.
     fn extract_esk(out_plaintext: &OutPlaintextBytes) -> Option<Self::EphemeralSecretKey>;
+
+    /// Parses the given note plaintext bytes.
+    ///
+    /// Returns `None` if the byte slice does not represent a valid note plaintext.
```

Comment:
This is ambiguous. "Valid" could mean having a recognized lead byte, for example. In fact it only checks the length.
```suggestion
    /// Returns `None` if the byte slice is of the wrong length for a note plaintext.
```